post-processor/googlecompute-export/post-processor.go

```diff
@@ -119,14 +169,37 @@ func (p *PostProcessor) Configure(raws ...interface{}) error {
 		p.config.Network = "default"
 	}
 
-	if p.config.AccountFile != "" && p.config.VaultGCPOauthEngine != "" {
-		errs = packersdk.MultiErrorAppend(
-			errs, fmt.Errorf("May set either account_file or "+
-				"vault_gcp_oauth_engine, but not both."))
+	err = p.CheckAuth()
+	if err != nil {
+		errs = packersdk.MultiErrorAppend(errs, err)
 	}
-	if p.config.AccountFile != "" && p.config.AccessToken != "" {
-		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("You cannot "+
-			"specify access_token and account_file at the same time"))
+
+	// Authenticating via an account file
+	if p.config.AccountFile != "" {
+		log.Printf("account_file is deprecated, please use either credentials_json or credentials_file instead")
+		// Heuristic, but should be good enough to discriminate between
+		// the two somewhat reliably.
+		if strings.HasPrefix(strings.TrimSpace(p.config.AccountFile), "{") {
```

Comment:
An alternative approach if you want to gain more confidence would be to create an anonymous struct and use json.Unmarshal to validate that one of the fields is populated or maybe just check that there is no unmarshal error. 

Using os.Stat to check it the string is a file is not a good option as it may lead to false positives if the file doesn't exist. So that's not an alternative. 

This is just a comment to provide an alternative not that it needs to be done.

Reply:
Yeah unmarshalling might be an idea as well indeed, I think however we're good with this heuristic.
It's less secure than the alternative as it's thrown it off with a filename that starts with `{`, but thankfully I'd think this is extremely rare, and in such cases, I would easily suggest the user moves to using `credentials_file` :)